packages/components/src/components/hds/form/key-value-inputs/index.ts

```diff
@@ -0,0 +1,179 @@
+        AddRowButton?: ComponentLike<HdsFormKeyValueInputsAddRowButtonSignature>;
+        Error?: WithBoundArgs<
+          typeof HdsFormErrorComponent,
+          'contextualClass' | 'controlId' | 'onInsert' | 'onRemove'
+        >;
+      },
+    ];
+  };
+  Element: HdsFormFieldsetSignature['Element'];
+}
+
+// @ts-expect-error: decorator function return type 'ClassOf<AriaDescribedBy>' is not assignable to 'typeof HdsFormField'
+@ariaDescribedBy
+export default class HdsFormKeyValueInputs extends Component<HdsFormKeyValueInputsSignature> {
+  private _id = guidFor(this);
+  private _element!: HTMLElement;
+  @tracked _gridTemplateColumns = '';
+
+  get canDeleteRow(): boolean {
+    return this.args.data.length > 1;
+  }
+
+  @action _setUpColumn(): void {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      this._updateColumns();
+    });
+  }
+
+  @action _removeColumn(): void {
+    // eslint-disable-next-line ember/no-runloop
+    schedule('afterRender', (): void => {
+      this._updateColumns();
+    });
+  }
+
+  @action
+  appendDescriptor(element: HTMLElement): void {
+    registerAriaDescriptionElement(this as AriaDescribedByComponent, element);
+  }
+
+  @action removeDescriptor(element: HTMLElement): void {
+    unregisterAriaDescriptionElement(this as AriaDescribedByComponent, element);
+  }
+
+  private _setUpKeyValueInputs = modifier((element: HTMLElement) => {
+    this._element = element;
+  });
+
+  // Update the column array based on how they are ordered in the DOM
+  private _updateColumns = () => {
+    const columns = this._element
+      .querySelector(KEY_VALUE_PAIR_FIRST_ROW_SELECTOR)
+      ?.querySelectorAll(
+        `${KEY_VALUE_PAIR_FIELD_SELECTOR}, ${KEY_VALUE_PAIR_GENERIC_SELECTOR}, ${KEY_VALUE_PAIR_DELETE_ROW_CONTAINER_SELECTOR}`
+      );
+
+    let newGridTemplateColumns = '';
+
+    columns?.forEach((column, index) => {
+      const columnElement = column as HTMLDivElement;
+
+      // do substring to remove the leading dot from the class selector
+      if (
+        column.classList.contains(KEY_VALUE_PAIR_FIELD_SELECTOR.substring(1))
+      ) {
+        if (columnElement.dataset['width']) {
+          newGridTemplateColumns += `${columnElement.dataset['width']} `;
+        } else {
+          newGridTemplateColumns += '1fr ';
+        }
+      }
+      // do substring to remove the leading dot from the class selector
+      if (
+        column.classList.contains(KEY_VALUE_PAIR_GENERIC_SELECTOR.substring(1))
+      ) {
+        newGridTemplateColumns += 'auto ';
+
+        columnElement.style.setProperty(
+          '--hds-key-value-pair-column-index',
+          `${index + 1}`
+        );
+      }
+
+      if (
+        column.classList.contains(
+          KEY_VALUE_PAIR_DELETE_ROW_CONTAINER_SELECTOR.substring(1)
+        )
+      ) {
+        newGridTemplateColumns += 'min-content ';
+
+        columnElement.style.setProperty(
```

Comment:
```suggestion
        // since we're setting a `grid-row` value via CSS, we also have to set the `grid-column` value for the `delete-button-container` slot
        columnElement.style.setProperty(
```